src/node_main_instance.cc

```diff
@@ -0,0 +1,160 @@
+#include "node_main_instance.h"
+#include "node_internals.h"
+#include "node_options-inl.h"
+#include "node_v8_platform-inl.h"
+
+namespace node {
+
+using v8::Context;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::Local;
+using v8::Locker;
+using v8::SealHandleScope;
+
+NodeMainInstance::NodeMainInstance(uv_loop_t* event_loop,
+                                   const std::vector<std::string>& args,
+                                   const std::vector<std::string>& exec_args)
+    : args_(args),
+      exec_args_(exec_args),
+      array_buffer_allocator_(CreateArrayBufferAllocator::Create()),
```

Comment:
I think this should be

```suggestion
      array_buffer_allocator_(ArrayBufferAllocator::Create().release()),
```

Reply:
@joyeecheung The `.release()` isn’t necessary if you switch to a plain `std::unique_ptr<>`, if you want

Reply:
Do you mean switching the type in the class member declaration? That's actually what I did before opening this PR, but then I changed to `DeleteFnPtr` to align with the worker implementation. Is it going away?

Reply:
BTW, I think I still need to do a `.release()` because I need to cast it along with `FreeArrayBufferAllocator()` as the deleter.

Reply:
um, wait, looks like I don't need the custom deleter, the two Free* are not more sophisticated than a `delete` call..

Reply:
:+1: Thanks for pointing out the asymmetry with `node_worker.cc`, I’ve opened https://github.com/nodejs/node/pull/27251 for that